sql/server.go

```diff
@@ -154,82 +168,73 @@ func (p parameters) Arg(i int) (parser.Datum, bool) {
 
 // exec executes the request. Any error encountered is returned; it is
 // the caller's responsibility to update the response.
-func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
-
-	// Pick up current session state.
-	// The request user is validated in ServeHTTP. Even in insecure mode,
-	// it is guaranteed not to be empty.
-	planMaker := planner{user: req.GetUser()}
-	defer func() {
-		// Update session state even if an error occurs.
-		if bytes, err := gogoproto.Marshal(&planMaker.session); err != nil {
-			// Marshaling a `Session` never errors (known from reading the code).
-			panic(err)
-		} else {
-			resp.Session = bytes
-		}
-	}()
+func (s *Server) exec(req driver.Request, planMaker *planner) driver.Response {
+	var resp driver.Response
 
-	if req.Session != nil {
-		// TODO(tschottdorf) will have to validate the Session information (for
-		// instance, whether access to the stored database is permitted).
-		if err = gogoproto.Unmarshal(req.Session, &planMaker.session); err != nil {
-			return
-		}
-	}
-	var stmts parser.StatementList
-	if stmts, err = parser.Parse(req.Sql); err != nil {
-		return
+	// TODO(vivek): This should parse each statement independently, report parse
+	// errors where necessary and execute all statements that were valid.
+	stmts, err := parser.Parse(req.Sql)
+	if err != nil {
+		// A parse error in one of the statements results in all the statements
+		// mistaken as one statement with a parse error.
+		var errProto proto.Error
+		errProto.SetResponseGoError(err)
+		resp.Results = append(resp.Results, driver.Result{Error: &errProto})
+		return resp
 	}
 	for _, stmt := range stmts {
-		// Bind all the placeholder variables in the stmt to actual values.
-		if err = parser.FillArgs(stmt, parameters(req.Params)); err != nil {
-			return
-		}
-		var plan planNode
-		if err = s.db.Txn(func(txn *client.Txn) error {
-			planMaker.txn = txn
-			plan, err = planMaker.makePlan(stmt)
-			planMaker.txn = nil
-			return err
-		}); err != nil {
-			return
-		}
+		var result driver.Result
```

Comment:
Its actually not present now with the move to using execStmt(). I do prefer the declaration versus assigning to an empty struct.